spec/SlowQuery.spec.js

```diff
@@ -0,0 +1,140 @@
+const delay = duration => new Promise(resolve => setTimeout(resolve, duration));
+describe('Slow Queries', () => {
+  it('can enable slow queries', async () => {
+    await reconfigureServer({
+      slowQuery: {
+        enable: true,
+        threshold: 300,
+        log: true,
+      },
+    });
+    Parse.Cloud.beforeSave('TestObject', async () => {
+      await delay(500);
+    });
+    const logger = require('../lib/logger').logger;
+    let call = '';
+    spyOn(logger, 'warn').and.callFake(warn => {
+      call = warn;
+    });
+    await new Parse.Object('TestObject').save();
+    expect(call.includes('Detected a slow query on path /classes/TestObject.')).toBeTrue();
+    await delay(1000);
+    const slowQuery = await new Parse.Query('_SlowQuery').first({ useMasterKey: true });
+    expect(slowQuery).toBeDefined();
+    expect(slowQuery.get('method')).toBe('POST');
+    expect(slowQuery.get('path')).toBe('/classes/TestObject');
+    expect(slowQuery.get('body')).toEqual({});
+    expect(slowQuery.get('query')).toEqual({});
+    expect(slowQuery.get('duration')).toBeGreaterThan(500);
+  });
+
+  it('needs masterKey for slow queries', async () => {
+    await reconfigureServer({
+      slowQuery: {
+        enable: true,
+        threshold: 300,
+        log: true,
+      },
```

Comment:
Is it possible to add a test to check that Parse.Cloud.beforeSave("_SlowQuery") works correctly. It could help developers to manipulate the body and remove sensitive data (we should add documentation suggestion about that)

I think we need to ensure/cover that this is supported.